runtime/parser2/statement.go

```diff
@@ -122,6 +122,46 @@ func parseStatement(p *parser) ast.Statement {
 	}
 }
 
+func parseFunctionDeclarationOrFunctionExpressionStatement(p *parser) ast.Statement {
+
+	startPos := p.current.StartPos
+
+	// Skip the `fun` keyword
```

Comment:
Would be great to make this a convention:

Describe the token that we are advancing with a comment above all `p.next()` calls.

Reply:
Good idea, done in 5be849739ff3adfba996aa39490bbc6c766e83c2